### PR TITLE
[Task#69294]Create Filter Salon

### PR DESCRIPTION
**Pull request by Hungnd-0370**
## Related Tickets:
- [#69294](https://edu-redmine.sun-asterisk.vn/issues/69294)

## WHAT this PR does?
-Tạo filter cho salon 

## Notes Impacted Areas:
- resources\js\Pages\salons\Salons.jsx

## Attachments:
- image

### CODE REVIEW

**Review by dangvm-0369 — APPROVED** [review]
oke

**Comment by Hungnd-0370** [comment]
ready

File: resources/js/Pages/salons/Salons.jsx [hunk 1]
```diff
@@ -31,14 +31,33 @@ export default function Salons(props) {
...
+                    value: '1',
+                },
+                {
+                    text: 'Medium',
+                    value: '2',
+                },
+                {
+                    text: 'Big',
+                    value: '3',
+                },
```
Comment (ghost) [inline resources/js/Pages/salons/Salons.jsx]: data sao fix cứng vậy?

**Comment by Hungnd-0370** [comment]
ready

File: database/factories/RegistrationFactory.php [hunk 2]
```diff
@@ -27,7 +27,7 @@ public function definition()
             'seats_number' => $this->faker->numberBetween(1,5000),
             'staffs_number' => $this->faker->numberBetween(1,5000),
             'package_id' => Package::all()->random()->id,
-            'status' => $this->faker->numberBetween(1,3),
+            'status' => $this->faker->numberBetween(0,2),
```
Comment (ghost) [inline database/factories/RegistrationFactory.php]:
```suggestion
            'status' => $this->faker->numberBetween(0, 2),
```

File: resources/js/Pages/salons/Salons.jsx [hunk 3]
```diff
@@ -67,7 +86,7 @@ export default function Salons(props) {
         if(valuePrev !== '' && value !== ''){
             const salonsSearched = [];
             props[0].salons.forEach( (salon) => {
-                if(salon.id == value || salon.name === value || salon.owner_email === value) salonsSearched.push(salon);
+                if(salon.id == value || salon.name.includes(value) || salon.owner_email.includes(value)) salonsSearched.push(salon);
```
Comment (ghost) [inline resources/js/Pages/salons/Salons.jsx]:
```suggestion
                if (salon.id == value
                    || salon.name.includes(value)
                    || salon.owner_email.includes(value)
                ) {
                    salonsSearched.push(salon);
                };
```

**Comment by Hungnd-0370** [comment]
ready

**Comment by Hungnd-0370** [comment]
ready

File: app/Http/Controllers/SalonController.php [hunk 4]
```diff
@@ -120,11 +120,11 @@ private function calculateSalon()
...
 
         foreach ($allSalons as $salon) {
-            $salonID = $salon->id;
+            $salon->package;
 
-            $numStaffs = count(Salon::find($salonID)->users);
+            $numStaffs = count($salon->users);
 
-            $numCustomers = count(Salon::find($salonID)->customers);
+            $numCustomers = count($salon->customers);
```
Comment (ghost) [inline app/Http/Controllers/SalonController.php]: n+1 queries?

**Comment by Hungnd-0370** [comment]
ready

**Pull request merged**